xarray/backends/common.py

```diff
@@ -178,6 +178,15 @@ def _normalize_path_list(
     return _normalize_path_list(paths)
 
 
+def _open_file(file, mode, storage_options=None):
+    import fsspec
+
+    fs, _, paths = fsspec.get_fs_token_paths(
+        file, mode=mode, storage_options=storage_options
+    )
+    return fs.open(paths[0], mode=mode)
```

Comment:
@dcherian I think this get's at your comment here https://github.com/pydata/xarray/issues/9723#issuecomment-2460464611 about restricting to a simple utility function using `fsspec`, but let me know if I'm missing something